xir/parser.py

```diff
@@ -192,10 +193,17 @@ def gate_def(self, args):
                 if has_declared_wires:
                     stmts = args[i:]
                     break
-
                 stmts.append(arg)
-                for w in arg.wires:
-                    wires += (w,) if w not in wires else ()
+
+                int_wires = [w for w in arg.wires if isinstance(w, int)]
+                wires += tuple(arg.wires)
+                if int_wires and max(int_wires) > max_wire:
+                    max_wire = max(int_wires)
+
+            if not has_declared_wires:
+                wires = tuple(range(max_wire + 1))
+            else:
+                wires = tuple(dict.fromkeys(wires))
```

Comment:
Is the purpose of this transformation to remove duplicate wires while maintaining their order? If so, that's pretty clever, but also deserves a comment.

Reply:
Yes, exactly. Will add a comment!